crates/milo/src/ark.rs

```
@@ -0,0 +1,32 @@
+use std::error::Error;
+use std::fs::File;
+
+use crate::fio::read_u32;
+use crate::traits::Load;
+pub mod amp;
+pub mod freq;
+
+pub enum ArkTypes {
+    Empty,
+    FreqArk(freq::FreqArchive),
+    AmpArk(amp::AmpArchive),
+}
+
+pub fn load_ark_file(f: &mut File) -> Result<ArkTypes, Box<dyn Error>> {
+    let vercheck = read_u32(f, true)?;
+    let mut ark: ArkTypes = ArkTypes::Empty;
+    match vercheck {
+        0x004B5241 => {
+            let mut freq = freq::FreqArchive::new();
+            freq.load(f, 0)?;
+            ark = ArkTypes::FreqArk(freq);
+        }
+        0..=2 => {
+            let mut amp = amp::AmpArchive::new();
+            amp.load(f, 0)?;
+            ark = ArkTypes::AmpArk(amp);
+        }
+        _ => println!("unrecognized ark version. if gh1 or later, use the .hdr and not the .ark")
+    }
+    Ok(ark)
+}
```

Instead of returning `ArkTypes::Empty` for an unrecognized ark, remove that enum variant and return an error instead.

This will make your actual intent much clearer.